pkg/alertmanager/amcfg.go

```diff
@@ -947,6 +960,42 @@ func (cb *configBuilder) convertPushoverConfig(ctx context.Context, in monitorin
 	return out, nil
 }
 
+func (cb *configBuilder) convertTelegramConfig(ctx context.Context, in monitoringv1alpha1.TelegramConfig, crKey types.NamespacedName) (*telegramConfig, error) {
+	out := &telegramConfig{
+		VSendResolved:        in.SendResolved,
+		APIUrl:               in.APIURL,
+		ChatID:               in.ChatID,
+		Message:              in.Message,
+		DisableNotifications: false,
+		ParseMode:            in.ParseMode,
+	}
+
+	if in.HTTPConfig != nil {
+		httpConfig, err := cb.convertHTTPConfig(ctx, *in.HTTPConfig, crKey)
+		if err != nil {
+			return nil, err
+		}
+		out.HTTPConfig = httpConfig
+	}
+
+	if in.ChatID == 0 {
+		return nil, errors.Errorf("mandatory field %q is empty", "chatID")
+	}
+
+	if in.BotToken != nil {
+		botToken, err := cb.store.GetSecretKey(ctx, crKey.Namespace, *in.BotToken)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get bot token")
+		}
+		if botToken == "" {
+			return nil, errors.Errorf("mandatory field %q is empty", "botToken")
+		}
```

Comment:
shouldn't this be checked in `sanitize`?

Reply:
Done!